src/simulators/matrix_product_state/matrix_product_state_internal.cpp

```diff
@@ -1510,20 +1517,30 @@ uint_t MPS::apply_measure_internal_single_qubit(uint_t qubit, RngEngine &rng) {
     measurement_matrix = measurement_matrix * (1 / sqrt(prob1));
   }
   apply_matrix_internal(qubits_to_update, measurement_matrix);
-  propagate_to_neighbors_internal(qubit, qubit);
+  propagate_to_neighbors_internal(qubit, qubit, measure_all);
+
   return measurement;
 }
 
-void MPS::propagate_to_neighbors_internal(uint_t min_qubit, uint_t max_qubit) {
+void MPS::propagate_to_neighbors_internal(uint_t min_qubit, uint_t max_qubit, 
+					  bool measure_all) {
+  uint_t right_qubit=num_qubits_-1;
+  int_t left_qubit=0;
+
+  if (measure_all) {
+    right_qubit = max_qubit >= num_qubits_-2 ? num_qubits_-1 : max_qubit + 1;
+    left_qubit = min_qubit == 0 ? 0 : min_qubit - 1;
+  }
+  if (num_qubits_ == 1)
+    right_qubit = 0;
```

Comment:
In the case of `measure_all`, since we are measuring in order from `0` to `n-1`, this will not really be necessary - there will be no propagation to the left. But I think it is better to cover this, for the sake of completeness. Let me know if you think I should do this differently.